### DISCUSSION TITLE
2d view

### DISCUSSION THREAD

**Discussion by MarinCronograph** [gempy-project/gempy#932] [General]
hello,

i am trying to get 2d view with this line. I have imported gempy_viewer and have created initial model extent (geo_model: gp.data.GeoModel = gp.create_geomodel....

Am using this line  to visualise  p2d = gpv.plot_2d(geo_model)  and getting warning . UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
  p.fig.show()

However if i change  to p2d = gpv.plot_3d(geo_model)  it shows axes that i can rotate (after installing pyvista)

HAve googled about this 2d warning and there was adviced to use -matplotlib.use('Agg')

However this didnt help. 

p.s I am  using Anaconda environment, Spyder , Python version 3.12 

rgds

MArin Mirosevic

**Comment by javoha** [comment]
Hi,
thanks for your question. What happens if you just run `gpv.plot_2d(geo_model)`? And do you use your own model or one of the examples?
If it is your own -can you try one of the simple examples ([like this](https://docs.gempy.org/examples/geometries/g07_combination.html#sphx-glr-examples-geometries-g07-combination-py)) and let us know if it works there?

Cheers
Jan

**Reply by MarinCronograph** [reply]
Hi,

i was following steps on link under

https://docs.gempy.org/tutorials/a_getting_started/get_started.html#sphx-glr-tutorials-a-getting-started-get-started-py


I have written as you said , in a separate file  , just  **gpv.plot_2d(geo_model)** ,  same notification

"UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
  p.fig.show()"
Out[1]: <gempy_viewer.modules.plot_2d.visualization_2d.Plot2D at 0x1858267d9a0>


**p.s python code in separate file i used to run as you said, under** **but same warning**

import gempy as gp
import gempy_viewer as gpv

geo_model: gp.data.GeoModel = gp.create_geomodel(
    project_name='Model1',
    extent=[0, 791, -200, 200, -582, 0],
    resolution=None,
    refinement=4, # We will use octrees
    structural_frame=gp.data.StructuralFrame.initialize_default_structure()
)

gpv.plot_2d(geo_model)


rgds

**Reply by javoha** [reply]
Ah ok, so I just reproduced your steps - I also get the warning, but the figure still shows. The first one is supposed to be an empty plot. I think the wanring is just to specify that this is not an interactive 3d plot, but evrything still works.

**Reply by MarinCronograph** [reply]
Beside warning I dont get any 2D view displayed, however 3d view works fine , it seems so.

Thanks.

**Reply by javoha** [reply]
This is really surprising. Could you please run the example I linked above and check if the error persists?

**Reply by MarinCronograph** [reply]
Thanks  I shall try on this link , https://docs.gempy.org/examples/geometries/g07_combination.html#sphx-glr-examples-geometries-g07-combination-py
